### DISCUSSION TITLE
Daily Email alerts not being sent

### DISCUSSION THREAD

**Discussion by its-a-me-mario-the-og** [Security-Onion-Solutions/securityonion#8393] [Unsupported Versions]
Hi I have a problem with my email alerting. The setup is a distributed setup with one manager and two forward nodes and one search node. The email alert file is in the /opt/so/rules/elastalert directory with the smtp auth file. The whole set up is on 2.3.130 and before .130 I was receiving daily email alerts. I was using port 25 but switched to port 587 but no change in the email situation. In the /opt/so/log/elastalert it shows daily alerts but nothing being sent out. I've attached the log file from elastalert.
[elastalert.log](https://github.com/Security-Onion-Solutions/securityonion/files/9221005/elastalert.log)

**Comment by dougburks** [comment]
From the log:
```
2022-07-28 15:07:52,700    ERROR           elastalert Error while running alert email: Error connecting to SMTP host: Connection unexpectedly closed
2022-07-28 15:16:06,185    ERROR           elastalert Error while running alert email: Error connecting to SMTP host: Connection unexpectedly closed
2022-07-28 15:25:58,920    ERROR           elastalert Error while running alert email: Error connecting to SMTP host: Connection unexpectedly closed
2022-07-28 15:34:04,594    ERROR           elastalert Error while running alert email: Error connecting to SMTP host: Connection unexpectedly closed
2022-07-28 15:44:08,083    ERROR           elastalert Error while running alert email: Error connecting to SMTP host: Connection unexpectedly closed
2022-07-28 15:52:19,666    ERROR           elastalert Error while running alert email: Error connecting to SMTP host: Connection unexpectedly closed
```

Are there any firewalls or any other devices that may interfere with traffic between your manager and the SMTP host?

From your manager, can you ping the SMTP host?

Can you connect to ports 25 and 587 using a tool like `telnet` or `nc`?

**Reply by its-a-me-mario-the-og** [reply]
We have tried using both 25 and 587 through telnet but no connection is coming back. We do have firewalls but checked the settings of the firewalls and both ports are open.

**Reply by dougburks** [reply]
If you can't telnet to 25 or 587, then it doesn't seem like a Security Onion problem.

**Reply by its-a-me-mario-the-og** [reply]
Correction we are getting a connection on 25 to the email server. 
```
Trying (removed)...
Connected to (removed).
Escape character is '^]'.
220 (removed) (IMail 12.5.8.58 265885-7) NT-ESMTP Server X1
ehlo
250-(removed) says hello
250-SIZE 26214400
250-8BITMIME
250-DSN
250-ETRN
250-AUTH LOGIN CRAM-MD5
250-AUTH LOGIN
250-AUTH=LOGIN
250 STARTTLS
quit
221 bye
Connection closed by foreign host
```

**Reply by dougburks** [reply]
To clarify, this connection to port 25 was from your Security Onion box?

What does your Elastalert log say now?

Have you double-checked the TLS and auth settings in your Elastalert config?

**Reply by its-a-me-mario-the-og** [reply]
Yes this was from my Security Onion Box, I'm adding in the updated elastalert log in the log their is something about SSL: SSLV3_ALERT_HANDSHAKE_FAILURE. I have looked it up in documentations for elastalert but nothing has helped. Below is what we have in the elastalert config file at this time and we have not changed anything in the config file since using Security Onion.

Elastalert config file:
rules_folder: /opt/elastalert/rules/
scan_subdirectories: true
disable_rules_on_error: false
run_every:
  minutes: 3
buffer_time:
  minutes: 10
old_query_limit:
  minutes: 5
es_host: removed
es_port: 9200
es_conn_timeout: 55
max_query_size: 5000
use_ssl: true
verify_certs: false
es_username: so_elastic
es_password: - removed
writeback_index: elastalert_status
alert_time_limit:
  days: 2
index_settings:
  shards: 1
  replicas: 0
logging:
  version: 1
  incremental: false
  disable_existing_loggers: false
  formatters:
    logline:
      format: '%(asctime)s %(levelname)+8s %(name)+20s %(message)s'
  handlers:
    file:
      class: logging.FileHandler
      formatter: logline
      level: INFO
      filename: /var/log/elastalert/elastalert.log
  loggers:
    '':
      level: INFO
      handlers:
      - file
      propagate: false
(END)

Elastalert log file:
[elastalert.log](https://github.com/Security-Onion-Solutions/securityonion/files/9337528/elastalert.log)

**Reply by dougburks** [reply]
Have you set `smtp_ssl: true` as shown at https://docs.securityonion.net/en/2.3/elastalert.html#configuration?

**Reply by its-a-me-mario-the-og** [reply]
Yes we have done that and still no change in the logs or receiving daily emails.
